pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -50,6 +50,9 @@ class TypeHintMatch:
     kwargs_type: str | None = None
     """kwargs_type is for the special case `**kwargs`"""
     has_async_counterpart: bool = False
+    """`function_name` and `async_function_name` share arguments and return type"""
+    compulsory: bool = False
```

Comment:
if this PR is reopened, I would suggest "mandatory" instead. Its a much more common term for these things.